content/kubeone/master/guides/addons/_index.en.md

```diff
@@ -81,9 +85,105 @@ the path is relative to the KubeOne configuration file. This means that
 
 {{% notice note %}}
 Subdirectories are not considered when applying addons. Only addons in the root
-of the provided directory will be applied.
+of the provided directory will be applied. E.g. `./addons/addon_name1` is
+supported addon while `./addons/groupping/addon_name1` and
+`./addons/addon_name1/subdir1` does not.
+{{% /notice %}}
+
+## Embedded Addons
+
+Since v1.3 release KubeOne has new kind of addons that always caried with the
+binariy itself, embedded addons. This is done thanks to new Go's
+[`embed`][embed-docs] package. 
+
+Some of those addons are directly and automatically used by the KubeOne itself
+(like machinecontroller for example). But it's not the case for all of them.
+Some very useful addons need to be activated by the user in order to be deployed
+to the cluster, like `backups-restic`, `cluster-autoscaler` or
+`unattended-upgrades`.
+
+### Activate Embedded Eddons
+
+Or order to activate such embedded addons, user need to use new Addons
+[API][addons-api].
+
+Example:
+```yaml
+apiVersion: kubeone.io/v1beta1
+kind: KubeOneCluster
+versions:
+  kubernetes: 1.20.1
+
+addons:
+  enable: true
+  path: "./addons"
+  addons:
+  - name: cluster-autoscaler
+  - name: unattended-upgrades
+```
+
+[Up to date (and possibly unreleased yet) list of addons.][addons-list-url]
+
+### Overriding Embedded Eddons
+
+Some of those embedded addons ad used to be written in Go structures form. Now
+KubeOne has them in YAML form so those addons are easier to maintain and to let
+users to overrider them.
+
+For example if you wish to have own machinecontroller manifests being deployed
+to the cluster, you cloud override `machinecontroller` addons. Enable addons
+like described above and in the addons directory place your machine-controller
+manifests in the same directory name, e.g.
+`./addons/machinecontroller/yourmanifest.yaml`.
+
+## Cleanup Addons
+
+There might be a situation when user might want to remove the addon
+declaratively. You will need to use new `delete` filed from Addons [API].
+
+```yaml
+apiVersion: kubeone.io/v1beta1
+kind: KubeOneCluster
+versions:
+  kubernetes: 1.20.1
+
+addons:
+  enable: true
+  path: "./addons"
+  addons:
+  - name: unattended-upgrades
+    delete: true
+```
+
+{{% notice note %}}
+That will cleanup all the traced of `unattended-upgrades` addon, but only it's
+manifests, not the contents that addons workloads might have produced, i.e.
+backups themselves.
 {{% /notice %}}
 
+## Parameters
+
+It's possible to pass down user defined parameters (in key/value form) down to
+an addon. There are two fields (of type map[string]string):
+`.addons.globalParams` and `.addons.addons[] .params` in each addon.
```

Comment:
It might look better with bullet-points. I'd also add a note such as this one.

```suggestion

* `.addons.globalParams`
* `.addons.addons[].params` in each addon.

{{% notice note %}}
`.addons.addons[].params` has higher priority over `.addons.globalParams`, so you can use it to override globally defined parameters.
{{% /notice %}}
```